ffi/cpp/include/fixed_decimal.hpp

```diff
@@ -0,0 +1,35 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#ifndef ICU4X_FIXED_DECIMAL_HPP
+#define ICU4X_FIXED_DECIMAL_HPP
+
+#include <algorithm>
+#include <memory>
+
+#include "../../capi/include/fixed_decimal.h"
+
+namespace icu4x {
+struct ICU4XFixedDecimalDeleter {
+  void operator()(ICU4XFixedDecimal* l) const noexcept {
+    icu4x_fixed_decimal_destroy(l);
+  }
+};
```

Comment:
> thought

Sometimes with things like this I like to pass a pointer reference `ICU4XFixedDecimal*&` and then also add an `l = nullptr;` (In the worst case, I'd rather have an invalid access to `nullptr` than a dangling pointer into other memory.)

This may not be an issue, since we are using `unique_ptr` and we're _supposed_ to have only one handle to the data.

Reply:
I'm picking the signature suggested by uintptr docs; I don't want overload resolution to get in the way here. This is a preexisting API I'm implementing, so the signature isn't mine to decide.

Reply:
Good to know! I haven't worked with `unique_ptr` this directly, which is why I suspected that this signature was probably what we do want. 

Thanks for explaining!